### PR TITLE
Extend understandimporter by class metrics

### PR DESCRIPTION
**Pull request by ukinimod**
# Extend understandimporter by class metrics

## Description

see #240 connects #240 

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

* [ ] **All** requirements mentioned in the issue are implemented
* [x] Does match the Code of Conduct and the Contribution file 
* [x] Task has its own **GitHub issue** (something it is solving)
  * [ ] Issue number is **included in the commit messages** for traceability
* [ ] **Update the README.md** with any changes/additions made
* [ ] **Update the CHANGELOG.md** with any changes/additions made
* [ ] **Enough test coverage to ensure that uncovered changes do not break functionality**
* [x] **All tests pass**    
* [ ] **Descriptive pull request text**, answering:
  + What problem/issue are you fixing?
  + What does this PR implement and how? 
* [x] **Assign your PR to someone** for a code review
  + This person _will be contacted **first**_ if a bug is introduced into `master`
* [ ] **Manual testing** did not fail

### CODE REVIEW

**Comment by ukinimod** [comment]
The PR introduces the following bugs:
- SourceMonitorImporter no longer working correctly as wrong value for path are read.
- UnderstandImporter and SourceMonitorImporter have metrics with empty names.

I suggest fixing these issues before merging.

**Comment by Richargh** [comment]
Ok, so who is working on this bug?

**Comment by Richargh** [comment]
@ukinimod Do you want help? We could prioritze it for next week.

**Comment by ukinimod** [comment]
@Richargh I fixed the issue. 

@mrpatpat Could you do the review?

File: analysis/import/UnderstandImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/understand/UnderstandProjectBuilder.kt [hunk 1]
```diff
@@ -108,15 +248,9 @@ class UnderstandProjectBuilder(
     private fun ProjectBuilder.insertRow(rawRow: Array<String?>, header: UnderstandCSVHeader) {
         try {
             val row = UnderstandCSVRow(rawRow, header, pathSeparator)
-            if (row.isFileRow) {
-                this.insertByPath(row.pathInTree(), row.asNode())
-            }
+            this.insertByPath(row.pathInTree(), row.asNode())
         } catch (e: IllegalArgumentException) {
-            logger.warn { "Ignoring row due to ${e.message}" }
+            logger.warn { "Ignoring $rowNumber-th row ${Arrays.toString(rawRow)} due to: ${e.message}" }
```
Comment (alschmut) [inline analysis/import/UnderstandImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/understand/UnderstandProjectBuilder.kt]: When running the build I get 3106 WARN messages like the one below about ignoring rows. Do these show up on purpose? 
```
[main] WARN de.maibornwolff.codecharta.importer.understand.UnderstandProjectBuilder - Ignoring 3106-th row [Function, selected.(unnamed_function_9), visualization/app/codeCharta/ui/legendPanel/legendPanelController.spec.js, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 1, null, 1, null, 0, null, 0, null, 1, 1, null, null, null, 0, null, 0, null, null, 1, 0, null, null, 0, 0, null, 1, 1, null, 1, 1, 1, 1, null, null, null, null, null, null, null, 0, null, null, null, null, null, null, null] due to: Kind Function not supported, yet.
```

**Comment by Richargh** [comment]
pull-request monkeyuser :)

Can you update the branch so it's current with master?

File: analysis/import/UnderstandImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/understand/UnderstandProjectBuilder.kt (hunk 1 above)
Comment (ukinimod) [reply analysis/import/UnderstandImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/understand/UnderstandProjectBuilder.kt]: I is on intension. Maybe a different log level (Info) would be ok, as it is an anticipated issue with a workaround.

**Comment by ukinimod** [comment]
@Richargh Any constructive suggestion for splitting up future prs is welcome.

@alschmut Any suggestions where the code could be more clear. Code review is (mostly) not about finding mistakes but improving coding style / readability / understanding... I have written the code for you, for everybody working on codecharta to read. There is the saying of Uncle Bob that code should read like well written prose. Is there a place where the code could be written nicer?

File: analysis/import/UnderstandImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/understand/UnderstandProjectBuilder.kt (hunk 1 above)
Comment (alschmut) [reply analysis/import/UnderstandImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/understand/UnderstandProjectBuilder.kt]: yea I think another log level is a good idea

**Comment by alschmut** [comment]
@ukinimod Thanks for your comment! Rather reviewing the code style than the code correctness actually makes sense to me. And I have to admit I like your code style!, I'm just not that into the ccsh as much as I am into the visualisation ;) Thats what still makes it more complex to me.

**Comment by FernandoES** [comment]
I have been analyzing the code and I find it clear and readable, I found no improvement to suggest. Thumbs up

**Pull request merged**